hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaManager.java

```diff
@@ -453,6 +465,9 @@ void reset() {
         // permit count can exceed allowed parallelization count.
         replicaSyncSemaphore.drainPermits();
         replicaSyncSemaphore.release(maxParallelReplications);
+        if (logger.isFinestEnabled()) {
```

Comment:
If there is a bad configuration, then it is better to throw an exception. Now you are at risk of loosing the problematic configuration completely and I have no idea if the system will actually behave correctly.